dask/bag/tests/test_bag.py

```diff
@@ -1164,16 +1165,52 @@ def test_zip(npartitions, hi=1000):
 
 @pytest.mark.parametrize("nin", [1, 2, 7, 11, 23])
 @pytest.mark.parametrize("nout", [1, 2, 5, 12, 23])
-def test_repartition(nin, nout):
+def test_repartition_npartitions(nin, nout):
     b = db.from_sequence(range(100), npartitions=nin)
     c = b.repartition(npartitions=nout)
-
     assert c.npartitions == nout
     assert_eq(b, c)
     results = dask.get(c.dask, c.__dask_keys__())
     assert all(results)
 
 
+@pytest.mark.parametrize(
+    "nin, nout",
+    [(1, 1), (2, 1), (5, 1), (1, 2), (2, 2), (5, 2), (1, 5), (2, 5), (5, 5),],
+)
+def test_repartition_partition_size(nin, nout):
+    b = db.from_sequence(range(1, 100), npartitions=nin)
+    total_mem = sum(b.map_partitions(total_mem_usage).compute())
+    c = b.repartition(partition_size=(total_mem // nout))
+    assert c.npartitions >= nout
+    assert_eq(b, c)
+
+
+def test_multiple_repartition_partition_size():
+    b = db.from_sequence(range(1, 100), npartitions=1)
+    total_mem = sum(b.map_partitions(total_mem_usage).compute())
+
+    c = b.repartition(partition_size=(total_mem // 2))
+    assert c.npartitions >= 2
+    assert_eq(b, c)
+
+    d = c.repartition(partition_size=(total_mem // 5))
+    assert d.npartitions >= 5
+    assert_eq(c, d)
+
+
+def test_repartition_partition_size_complex_dtypes():
+    import numpy as np
```

Comment:
Since NumPy isn't a required dependency for Dask bag, we'll want to skip this test is NumPy isn't installed. Pytest has a convenient `pytest.importorskip` function for this. Here's an example of it in use:

https://github.com/dask/dask/blob/69a647ff2c80941a71afc11fadd76dbdddba48de/dask/bag/tests/test_bag.py#L369-L376